src/frontend/src/pages/TeamsPage/TeamMembersPageBlock.tsx

```diff
@@ -17,48 +17,56 @@ import PageBlock from '../../layouts/PageBlock';
 import DetailDisplay from '../../components/DetailDisplay';
 import NERSuccessButton from '../../components/NERSuccessButton';
 import NERFailButton from '../../components/NERFailButton';
+import NERAutocomplete from '../../components/NERAutocomplete';
+import { useToast } from '../../hooks/toasts.hooks';
 
 interface TeamMembersPageBlockProps {
   team: Team;
 }
 
-const userToAutocompleteOption = (user: User): { label: string; id: number } => {
-  return { label: `${fullNamePipe(user)} (${user.email})`, id: user.userId };
+const userToAutocompleteOption = (user: User): { label: string; id: string } => {
+  return { label: `${fullNamePipe(user)} (${user.email})`, id: `${user.userId}` };
 };
 
 const TeamMembersPageBlock: React.FC<TeamMembersPageBlockProps> = ({ team }) => {
   const auth = useAuth();
   const [isEditingMembers, setIsEditingMembers] = useState(false);
   const [members, setMembers] = useState(team.members.map(userToAutocompleteOption));
+  const [head, setHead] = useState(userToAutocompleteOption(team.head));
 
-  const { isLoading, isError, error, data: users } = useAllUsers();
-  const {
-    isLoading: setTeamMemberIsLoading,
-    isError: setTeamMemberIsError,
-    error: setTeamMemberError,
-    mutateAsync
-  } = useSetTeamMembers(team.teamId);
+  const { isLoading: allUsersIsLoading, isError: allUsersIsError, error: allUsersError, data: users } = useAllUsers();
+  const { isLoading: setTeamMembersIsLoading, mutateAsync: setTeamMembersMutateAsync } = useSetTeamMembers(team.teamId);
+  const { isLoading: setTeamHeadIsLoading, mutateAsync: setTeamHeadMutateAsync } = useSetTeamHead(team.teamId);
 
-  if (isError) return <ErrorPage message={error?.message} />;
-  if (setTeamMemberIsError) return <ErrorPage message={setTeamMemberError?.message} />;
-  if (isLoading || setTeamMemberIsLoading || !users) return <LoadingIndicator />;
+  const toast = useToast();
+
+  if (allUsersIsError) return <ErrorPage message={allUsersError?.message} />;
+  if (allUsersIsLoading || setTeamMembersIsLoading || setTeamHeadIsLoading || !users) return <LoadingIndicator />;
 
   const handleSubmit = async () => {
     try {
-      await mutateAsync(members.map((member) => member.id));
+      await setTeamMembersMutateAsync(members.map((member) => parseInt(member.id)));
+      await setTeamHeadMutateAsync(parseInt(head.id));
```

Comment:
I'm getting that same error you got where you can set someone who is a member as the head, and then everytime you try to set someone as head you get the (can't set member as team head)

Reply:
we're getting this error because you check this in edit team members, but not in the setTeamHead endpoint. Can you update the setTeamHead service method to remove someone as a member before we promote them to a head if they're a member of the current team